hashtable/hashtable.py

```diff
@@ -10,69 +10,74 @@ def __init__(self, key, value):
 
 
 class HashTable:
-    """
-    A hash table that with `capacity` buckets
-    that accepts string keys
-
-    Implement this.
-    """
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.storage = [None] * capacity
 
-    def fnv1(self, key):
-        """
-        FNV-1 64-bit hash function
-
-        Implement this, and/or DJB2.
-        """
+    def hash(self, key):
+        return hash(key)
 
     def djb2(self, key):
-        """
-        DJB2 32-bit hash function
+        hash = 5381
+
+        for x in key:
+            hash = (hash * 33) + ord(x)
 
-        Implement this, and/or FNV-1.
-        """
+        return hash
 
     def hash_index(self, key):
-        """
-        Take an arbitrary key and return a valid integer index
-        between within the storage capacity of the hash table.
-        """
-        #return self.fnv1(key) % self.capacity
         return self.djb2(key) % self.capacity
 
     def put(self, key, value):
-        """
-        Store the value with the given key.
-
-        Hash collisions should be handled with Linked List Chaining.
-
-        Implement this.
-        """
+        index = self.hash_index(key)
+        node = self.storage[index]
+
+        if node is None or node.key == key:
```

Comment:
I would check out this line and think about overwriting values. If you're given a node.key that is the same as the key being passed in on line 42, would you want to place a new node with the same key and a different value next?